packages/odyssey-react-mui/src/Checkbox.tsx

```diff
@@ -82,7 +87,13 @@ const Checkbox = ({
   value,
 }: CheckboxProps) => {
   const { t } = useTranslation();
-  const [isCheckedValue, setIsCheckedValue] = useState(isDefaultChecked);
+  const [isCheckedValue, setIsCheckedValue] = useState(isChecked || isDefaultChecked);
+
+  useEffect(() => {
+    if (typeof isChecked !== 'undefined') {
+      setIsCheckedValue(isChecked);
+    }
```

Comment:
```suggestion
    if (isChecked !== undefined) {
      setIsCheckedValue(isChecked);
    }
```

I think `typeof === 'undefined'` was used in the past because you could reassign the value of `undefined` in ES3. Not anymore!